### ISSUE TITLE
Feature Request: Better display lyrics

### ISSUE THREAD

**Issue by tisfeng** [kartik-venugopal/aural-player#111]
Hello, aural-player is a very powerful player, and I really like its design style. Thank you for your development!

I noticed that aural-player has a feature to display lyrics, but currently, the layout of the lyrics is not good, and it does not support dynamic lyrics display.

I am used to listening to songs while looking at the lyrics because some songs have amazing lyrics, and the lyrics are an inseparable part of the song.

I hope aural-player has a better way to conveniently display the lyrics, rather than placing them under the song information. It would be even better if it could support dynamic lyrics.

<img width="1667" alt="image" src="https://github.com/user-attachments/assets/d350a49a-194d-451a-a0f8-b2dbadf3fc52">

**Comment by kartik-venugopal** [comment]
Hi @tisfeng , thanks very much for the request.

Yes, I agree, it would be good to display dynamic lyrics, and I have thought about this in the past, but have not been able to find a song with dynamic lyrics, so I don't know what the format is like.

I understand that this may be difficult, but for the sake of development of this feature, are you able to share with me a song sample that has dynamic lyrics ? Or do you know where I can get such a file for free ?

Thanks!

**Comment by tisfeng** [comment]
Of course, the common lyric format used in songs is [LRC_(file_format)](https://en.wikipedia.org/wiki/LRC_(file_format)), and most music software supports displaying this format of lyrics.

As for the source of the lyric file lrc, some music apps will embed the lrc file when downloading music, while others may provide a separate lrc lyric file download with the song's name.

If the music app does not provide a lyric file, we can even make one ourselves, as its format is very simple.

Here is an example song [Fools](https://music.163.com/#/song?id=28442983)

song: [Anna F.-Fools.mp3.zip](https://github.com/user-attachments/files/17673493/Anna.F.-Fools.mp3.zip)
lyric:  [Anna F.-Fools.lrc.zip](https://github.com/user-attachments/files/17673495/Anna.F.-Fools.lrc.zip)

**Comment by tisfeng** [comment]
I haven't looked at the code yet. Aural seems to read the embedded lyrics in the songs, which is a simple method. However, for the songs we usually download from the internet, they typically do not come with lyrics, so an additional .lrc file would be needed.

I found a website for online searching of lrc files: https://www.toomic.com/?search=fools, which makes it easy to download lyrics.

Currently, Aural only supports playing local music, so we only need to consider the two situations mentioned above. If we later want to support playing online songs, we can use this method to display the lyrics.

<img width="1061" alt="image" src="https://github.com/user-attachments/assets/39c4356e-eeb5-4762-98b0-0cbb3f850bdf">

**Comment by kartik-venugopal** [comment]
Thanks very much for the info. One more question - where would you like to see the lyrice displayed ? In a separate window ?

I will try to get this implemented soon, but I have a lot of pending work for the v4 release and I'm a one man army, so hope you understand that it may take some time.

Cheers! 😄

**Comment by tisfeng** [comment]
> where would you like to see the lyrics displayed ? In a separate window ?

This sounds great. I noticed that Aural has some independent function windows, such as Play Queue and Effects, which users can choose to display freely. This design is fantastic.

I like this open-source lyrics project [LyricsX](https://github.com/MxIris-LyricsX-Project/LyricsX). It allows for easy lyric searching and dynamic lyric display. Hope this helps you.

I am looking forward to the lyrics feature. Thank you for your great work on Aural 🤗

<img width="896" alt="image" src="https://github.com/user-attachments/assets/b7216430-c70c-46e5-aa7b-43265cf06abd">

**Comment by tisfeng** [comment]
Hi, I've implemented the dynamic lyrics feature in a basic way. If you haven't done it yet, you can assign it to me.

I'm not familiar with the aural code, so it might not be perfect, but it runs. It seems okay, and you can optimize it later.

<img width="1282" alt="image" src="https://github.com/user-attachments/assets/c5106379-816f-4eea-9adb-71f88d5fdb66" />

**Comment by kartik-venugopal** [comment]
Hi @tisfeng , that's great, thank you! Yes, I'm happy to include your code. Can you send me a pull request ? I can merge and add the finishing touches.

**Comment by tisfeng** [comment]
No problem, give me some time, I need to organize the code, and I will submit a PR later.

**Comment by kartik-venugopal** [comment]
@tisfeng - Implemented and released in preview24.

- Did not use the SwiftUI view ... developed an AppKit view that matches the app's look n feel and is available > macOS Big Sur
- Implemented synchronized lyrics using polling (timer) ... seems to be no less CPU-intensive than your code.
- Impleme
- Added preferences for online search, lyrics directory, auto-scroll, karaoke mode, and auto window popup on track change.
- Added the ability to remove downloaded online lyrics (for cases when lyrics are inaccurate).
- Did not implement translations ... do not want to spend the time on this right now ... maybe later.

Cheers!
